tensorboard/plugins/graph/tf_graph_common/hierarchy.ts

```diff
@@ -884,13 +900,20 @@ function groupSeries(
     // If the series contains less than the threshold number of nodes and
     // this series has not been adding to the series map, then set this
     // series to be shown ungrouped in the map.
-    if (nodeMemberNames.length < threshold && !(seriesNode.name in map)) {
-      map[seriesNode.name] = tf_graph.SeriesGroupingType.UNGROUP;
+    if (
+      nodeMemberNames.length < threshold &&
+      hierarchy.getSeriesGroupType(seriesNode.name) ===
+        tf_graph.SeriesGroupingType.GROUP
```

Comment:
Previous code reads `!(seriesNode.name in map)` and it makes me think that this code is basically checking to see if the `hierarchy.getSeriesGroupType(seriesNode.name)` has the default value in case it does not exist. I think it is more prudent to `null` to `Map.prototype.has` than matching against the default value defined in various places in this module.

Would it not read better if the `hierarchy.getSeriesGroupType(seriesNode.name)` were to return `null`?

Reply:
I would still prefer to hide the fact that the group types are stored implicitly, because it is a concern that is specific to `Hierarchy` internals.  External consumers of the `Hierarchy` actually do not need to know what `null` values mean, so exposing only Group/Ungroup keeps consumers less complex and easier to reason about, imo.

While the previous code does check whether a node has been explicitly grouped/ungrouped, the new code matches the same intent (if a grouped series is too short, then ungroup it), and does not require readers to understand anything about the default group type.

I've updated the comment above on L900 accordingly.